src/test/java/com/google/research/bleth/simulator/GlobalResolverTest.java

```diff
@@ -0,0 +1,254 @@
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).isEmpty();
+    }
+
+    @Test
+    public void secondEstimateWithoutInformation_DoNotUpdateTheBoard() {
+        RealBoard realboard = new RealBoard(3, 3);
+        Beacon beacon = createRandomBeaconOnLocation(ZERO_ON_ZERO_COORDINATE, realboard);
+        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 3,
+                                                                /* colsNum= */ 3,
+                                                                ImmutableList.of(beacon));
+        resolver.receiveInformation(TWO_ON_TWO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        resolver.estimate();
+
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).containsExactly(beacon);
+    }
+
+    @Test
+    public void secondEstimateForUnlocatedBeacon_UpdateAccordingToTheObserverLocationOnly() {
+        RealBoard realboard = new RealBoard(3, 3);
+        Beacon beacon = createRandomBeaconOnLocation(ZERO_ON_ZERO_COORDINATE, realboard);
+        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 3,
+                                                                /* colsNum= */ 3,
+                                                                ImmutableList.of(beacon));
+        resolver.estimate();
+
+        resolver.receiveInformation(TWO_ON_TWO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).containsExactly(beacon);
+    }
+
+    @Test
+    public void transmissionsFromPreviousRoundsIsDeleted() {
+        RealBoard realboard = new RealBoard(5, 5);
+        Beacon beacon = createRandomBeaconOnLocation(ONE_ON_ZERO_COORDINATE, realboard);
+        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 5,
+                                                                /* colsNum= */ 5,
+                                                                ImmutableList.of(beacon));
+        resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        resolver.receiveInformation(new Location(4, 4), ImmutableList.of(beacon.transmit()));
+        resolver.estimate();
+
+        assertThat(resolver.getBoard().getAgentsOnLocation(TWO_ON_TWO_COORDINATE)).containsExactly(beacon);
+    }
+
+    private Beacon createRandomBeaconOnLocation(Location initialLocation, RealBoard realBoard) {
+        return BEACON_FACTORY.createBeacon(initialLocation, new RandomMovementStrategy(), realBoard);
+    }
+
+    private boolean isBoardEmpty(Board board, int rowsNum, int colsNum) {
```

Comment:
If this method is for testing only, it's a bit too limiting (can only test a single state) + produces a rather weird-looking test (`assertThat(isBoardEmpty(...)).isTrue()`). The error message you'd get on failure would also be rather uninformative.

Consider instead providing a method like `agentsOnBoard(Board)`, which would collect all agents from the board and return them as a `Map<Location, Agent>`; you'd then have assertions like `assertThat(agentsOnBoard(...)).isEmpty()`, or `assertThat(agentsOnBoard(...)).containsExactly(...)` etc. - it could simplify nearly all of your testcases.

Reply:
This method might be useful in another context as well: when writing the state of a Board object to the db we must iterate over all cells and retrieve the agents locations', so they could be recorded and written to the db.

Having a `agentsOnBoard()` method in `AbstractBoard` class will make the implementation of the `writeBoardState()` method more readable.